tools-local/unix/tool-helper.sh

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env bash
+
+# Provides helper functions.
+
+# Gets name of the operating system.
+# Exit 1 if unable to get operating system name.
+get_os_name()
+{
+    osName="$(uname -s)"
+
+    if [ $? -ne 0 ] || [ -z "$osName" ]; then
+        echo "Unable to determine the name of the operating system."
+        exit 1
+    fi
+
+    if echo "$osName" | grep -iqF "Darwin"; then
+        osName="OSX"
+    else
+        osName="Linux"
+    fi
+
+    echo "$osName"
+}
+
+# Gets the path to the folder corresponding to the specified tool name in Tools/downloads folder under repository root.
+# Exit 1 if unable to determine the path.
+get_repository_tools_downloads_folder()
+{
+    if [ -z "$1" ]; then
+        echo "Argument passed as repository-root is empty. Please provide a non-empty string."
+        usage
+        exit 1
+    fi
+
+    if [ -z "$2" ]; then
+        echo "Argument passed as tool name is empty. Please provide a non-empty string."
+        exit 1
+    fi
+
+    repoRoot="$1"
+    toolName="$2"
+    toolsPath="$repoRoot/Tools/downloads"
```

Comment:
I think we want to avoid the duplication of the Tools/downloads path.  I should be defined in a single place.  Right now it is specified in multiple places within .toolsversions